Source/Core/VideoCommon/XFMemory.h

```diff
@@ -17,75 +21,164 @@ constexpr size_t NUM_XF_COLOR_CHANNELS = 2;
 // Lighting
 
 // Projection
-enum : u32
+enum class TexSize : u32
+{
+  ST = 0,
+  STQ = 1
+};
+template <>
+struct fmt::formatter<TexSize> : EnumFormatter<TexSize::STQ>
 {
-  XF_TEXPROJ_ST = 0,
-  XF_TEXPROJ_STQ = 1
+  formatter() : EnumFormatter({"ST (2x4 matrix)", "STQ (3x4 matrix)"}) {}
 };
 
 // Input form
-enum : u32
+enum class TexInputForm : u32
+{
+  AB11 = 0,
+  ABC1 = 1
+};
+template <>
+struct fmt::formatter<TexInputForm> : EnumFormatter<TexInputForm::ABC1>
+{
+  formatter() : EnumFormatter({"AB11", "ABC1"}) {}
+};
+
+enum class NormalCount : u32
+{
+  None = 0,
+  Normals = 1,
+  NormalsBinormals = 2
+};
+template <>
+struct fmt::formatter<NormalCount> : EnumFormatter<NormalCount::NormalsBinormals>
 {
-  XF_TEXINPUT_AB11 = 0,
-  XF_TEXINPUT_ABC1 = 1
+  formatter() : EnumFormatter({"None", "Normals only", "Normals and binormals"}) {}
 };
 
 // Texture generation type
-enum : u32
+enum class TexGenType : u32
 {
-  XF_TEXGEN_REGULAR = 0,
-  XF_TEXGEN_EMBOSS_MAP = 1,  // Used when bump mapping
-  XF_TEXGEN_COLOR_STRGBC0 = 2,
-  XF_TEXGEN_COLOR_STRGBC1 = 3
+  Regular = 0,
+  EmbossMap = 1,  // Used when bump mapping
+  ColorSTRGBC0 = 2,
+  ColorSTRGBC1 = 3
+};
+template <>
+struct fmt::formatter<TexGenType> : EnumFormatter<TexGenType::ColorSTRGBC1>
+{
+  static constexpr array_type names = {
+      "Regular",
+      "Emboss map (used when bump mapping)",
+      "Color STRGBC0",
+      "Color STRGBC1",
+  };
+  formatter() : EnumFormatter(names) {}
 };
```

Comment:
I've just realized that `STRGBC0` is not supposed to be parsed as "STR" "GBC" "0" (store gbc 0) but instead "ST" "RGB" "C0" (Possibly still store, but I think it's actually the normal ST, using color 0's RGB).  It's also a bit odd since there currently is an assumption that it's only used in AB11 (in the software renderer), so only the R and G components are used.  Maybe it should be named just `Color0` or `Color0RG`.